src/plugins/intel_gpu/src/graph/reduce.cpp

```diff
@@ -95,6 +97,117 @@ layout reduce_inst::calc_output_layout(reduce_node const& node, kernel_impl_para
+        input0_layout.get<ShapeType>(),
+        ShapeType{0}
+    };
+
+    // output_shapes
+    std::vector<ShapeType> output_shapes = {ShapeType()};
+
+    // const_data
+    auto axes = desc->axes;
+    auto axes_tensor = std::make_shared<ngraph::runtime::HostTensor>(ov::element::i64, ov::Shape{axes.size()}, axes.data());
+    std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>> const_data = {{1, axes_tensor}};
+
+    // shape infer by mode
+    auto mode = desc->mode;
+    switch (mode) {
+        case reduce_mode::max:
+        {
+            ov::op::v1::ReduceMax op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::min:
+        {
+            ov::op::v1::ReduceMin op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::mean:
+        {
+            ov::op::v1::ReduceMean op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::prod:
+        {
+            ov::op::v1::ReduceProd op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::sum:
+        {
+            ov::op::v1::ReduceSum op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::logical_and:
+            {
+            ov::op::v1::ReduceLogicalAnd op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::logical_or:
+        {
+            ov::op::v1::ReduceLogicalOr op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::l1:
+        {
+            ov::op::v4::ReduceL1 op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::l2:
+        {
+            ov::op::v4::ReduceL2 op;
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::sum_square:
+            // not implemented
+        case reduce_mode::log_sum:
+            // not implemented
+        case reduce_mode::log_sum_exp:
+            // not implemented
+        default:
+            CLDNN_ERROR_MESSAGE(
+                desc->id,
+                "Not supported mode");
+    }
+
+    // get 'output_type'
+    auto output_type = input0_layout.data_type;
+    std::vector<reduce_mode> reduce_bool_modes = {reduce_mode::logical_and, reduce_mode::logical_or};
+    if (std::find(reduce_bool_modes.begin(), reduce_bool_modes.end(), mode) != reduce_bool_modes.end())
+        output_type = data_types::i8;
+    else if (output_type == data_types::i8 || output_type == data_types::u8)
+        output_type = data_types::f32;
```

Comment:
By the way @vladimir-paramuzov  this was from original calc_output_layout, do you know why it is so? Just for preventing overflow?

Reply:
yes. quantized operations by default saves results to FP32/16 precision unless there is a fused requantization. 
It works in a similar way for conv/deconv/fc etc

Reply:
Though the way how it's written is a bit confusing. 
@kelvinchoi-intel could you refactor condition to something like
```
else if (input_type== data_types::i8 || input_type == data_types::u8)
```

Reply:
Done